array.c

```diff
@@ -1142,6 +1142,9 @@ rb_ary_s_new(int argc, VALUE *argv, VALUE klass)
  *    a = Array.new(3)
  *    a # => [nil, nil, nil]
  *
+ *  Argument +size+, if given, must be an
+ *  {integer-convertible object}[rdoc-ref:Array@Integer-Convertible+Arguments].
```

Comment:
@jeremyevans, I'd like to hear from you about whether we should be doing this;  if so, there will be many such, both here in Array and elsewhere.